test/cljam/io/bam/decoder_test.clj

```diff
@@ -0,0 +1,143 @@
+      {:Xh {:type "H", :value (map unchecked-byte [0x1A 0xE3 0x01])}}
+      {:Xc {:type "B", :value "c,-128,0,127"}}
+      {:XC {:type "B", :value "C,0,127,255"}}
+      {:Xs {:type "B", :value "s,-32768,0,32767"}}
+      {:XS {:type "B", :value "S,0,32767,65535"}}
+      {:Xi {:type "B", :value "i,-2147483648,0,2147483647"}}
+      {:XI {:type "B", :value "I,0,2147483647,4294967295"}}
+      {:Xf {:type "B", :value "f,-0.3,0.0,0.3"}})
+    '({:CG {:type "B", :value "I,12,4,34,3"}}
+      {:Xa {:type "A", :value \p}}
+      {:Xi {:type "i", :value -100}}
+      {:Xf {:type "f", :value -1400.0}}
+      {:Xz {:type "Z", :value "AATTGGCC"}}
+      {:Xh {:type "H", :value (map unchecked-byte [0x1A 0xE3 0x01])}}
+      {:Xc {:type "B", :value "c,-128,0,127"}}
+      {:XC {:type "B", :value "C,0,127,255"}}
+      {:Xs {:type "B", :value "s,-32768,0,32767"}}
+      {:XS {:type "B", :value "S,0,32767,65535"}}
+      {:Xi {:type "B", :value "i,-2147483648,0,2147483647"}}
+      {:XI {:type "B", :value "I,0,2147483647,4294967295"}}
+      {:Xf {:type "B", :value "f,-0.3,0.0,0.3"}})
+
+    '({:MD {:type "Z" :value "2G0C0"}}
+      {:XT {:type "A", :value \R}}
+      {:NM {:type "i", :value 44}}
+      {:cg {:type "B", :value "I,3,1"}})
+    '({:MD {:type "Z" :value "2G0C0"}}
+      {:XT {:type "A", :value \R}}
+      {:NM {:type "i", :value 44}}
+      {:cg {:type "B", :value "I,3,1"}}
+      {:CG {:type "B", :value "I,34,0"}})
+
+    '()
+    '({:CG {:type "B", :value "I,34,7"}})))
+
+(deftest two-arguments-decode-alignment-test
+  (let [encoded-sam-byte-without-CG-tag
+        [0 0 0 0 8 0 0 0 5 30 73 18 2 0 0 0 6 0 0 0 -1 -1 -1 -1 -1 -1 -1 -1 0 0
+         0 0 114 48 48 51 0 85 0 0 0 96 0 0 0 20 40 17 -1 -1 -1 -1 -1 -1 88 88
+         66 83 4 0 0 0 17 49 2 0 20 0 112 0]
+        encoded-sam-byte-including-CG-tag
+        [0 0 0 0 6 0 0 0 5 30 73 18 2 0 -93 0 19 0 0 0 0 0 0 0 36 0 0 0 39 0 0
+         0 114 48 48 49 0 52 1 0 0 3 1 0 0 -120 20 24 17 20 20 65 -127 40 64
+         -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 67 71 66 73
+         10 0 0 0 8 0 0 0 0 0 0 0 4 0 0 0 1 0 0 0 4 0 0 0 0 0 0 0 1 0 0 0 2
+         0 0 0 3 0 0 0 0 0 0 0]]
+    (testing "two-arguments-decode-alignment-test"
+      (are [expected-aln byte]
+           (= expected-aln
+              (decoder/decode-alignment
+               [{:name "ref", :len 0}]
+               {:data (byte-array (map unchecked-byte byte))}))
+        (test-common/to-sam-alignment
+         {:qname "r003" :flag 0, :rname "ref", :pos 9, :end 14, :mapq 30,
+          :cigar "5H6M", :rnext "*", :pnext 0, :tlen 0, :seq "AGCTAA",
+          :qual "*" :options [{:XX {:type "B", :value "S,12561,2,20,112"}}]})
+        encoded-sam-byte-without-CG-tag
+
+        (test-common/to-sam-alignment
+         {:qname "r001", :flag 163, :rname "ref", :pos 7, :end 22, :mapq 30,
+          :cigar "8M4I4M1D3M", :rnext "=", :pnext 37, :tlen 39,
+          :seq "TTAGATAAAGAGGATACTG", :qual "*", :options ()})
```

Comment:
Extract the difference you want to check and put the common data to somewhere else.